test/fixtures/custom-properties.js

```diff
@@ -0,0 +1,11 @@
+import styled from 'astroturf';
+
+const FancyDiv = styled('div')`
+  color: ${props => (props.primary ? 'palevioletred' : 'red')};
```

Comment:
like, yes there are few things we can do like:

```
color: ${props => props.color}
```

but i wonder if it's really better than the v0.4.x API around dynamic stuff per https://github.com/4Catalyzer/astroturf/tree/v0.4.8#experimental-component-api

i mean, we could even combine that with the decss-style API (or maybe replace the decss-style thing entirely?)... something like:

```
color: green;

.${props => props.redness > 10} {
  color: red;
}
```

Reply:
the nice thing is it gives you a clear way to do dynamic values from JS, it's cleaner than the class approach for theme values etc. I think it doesn't replace the decss api if only b/c we need something for ie11... it's also a lot nicer for doing big chunks of style. I think they'd work well together depending on need

Reply:
also we already had this API: 

```
color: red;

${props => props.primary && css`
  background: white;
  color: palevioletred;
`}
```

like styled components for doing conditional classes...

Reply:
yeah... prettier sort of mangles that though. see https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBnGBPANnAJgCgHIAzCCQgSgAMAdKAAnsiwgCcl7W8BuOuxgCTAADqwjDU9ALwA+eqPGoAdKICWAWwCGrDPQBkepqlS0GjegCNNYANYBzMQFcouDgHcAFqvi8zjZmwcwpo4AG6qEDgwXLi+jFQAvvz0QgoS0nJpyhAwHnCs+oZgxqbmltb2Ti7uXj7J-pGB8iFw4ZFw0TzJiXRUvnTo2HhEpOTUyQHsnF1QyUqpYumy8ovKalo6CfTA9eW2DhDOrvSe3nBx5pNBLW1RMRdJZvMiqxkriiqsGtoYWzt+e0qh2qJ1q512V2aYQidxmjEefRAABoQOIYBEoKhkKBtGI3AAFbQILEoTShCCqXDIkAWVgVDoAZWCYFUUDsyGijjgKNZqHyMHxdLsWmQxBCfJRACtUAAPABCdNsjM06jgABlWXBReLuSBgqw+axkDTNBZsNBqWpYAB1Sm5ZAADgADCiFHzrXThMbRHBDaEtSiuABHRyqLiCzTCzTarAStBwdSqDmsLko1Csuw4ACKjhyWqQYtjupgpttuHtSAATCjoppVFgMwBhCDqEUoKDQAMgRx8gAqppJhb5CQSQA

Reply:
i guess this doesn't get around the issue of having to delete "invalid" props

Reply:
Tbf the valid prop check is minimal. It's more annoying in the non host component case tho, where you end up with props specifically for styles that get passed to Button or whatever. How do emotion and company deal with that?

Reply:
apparently you get an explicit hook: https://emotion.sh/docs/styled#customizing-prop-forwarding